pythia/utils/general.py

```diff
@@ -204,3 +198,20 @@ def get_current_tensors():
                 print(type(obj), obj.size())
         except:
             pass
+
+def find_complete_inds(decoder, next_word_inds):
+    incomplete_inds = []
+    for ind, next_word in enumerate(next_word_inds):
+        if next_word != decoder.vocab.EOS_INDEX:
+            incomplete_inds.append(ind)
+    complete_inds = list(set(range(len(next_word_inds))) - set(incomplete_inds))
+    return complete_inds, incomplete_inds
+
+def update_data(data, prev_word_inds, next_word_inds, incomplete_inds):
```

Comment:
Move this to your own decoder. This isn't general enough to be here.

Reply:
Okay. Will make this change along with the base class change.